misc/identity-keys/src/identity-keys.ts

```diff
@@ -0,0 +1,162 @@
+import { Store } from "@walletconnect/core";
+import {
+  JwtPayload,
+  composeDidPkh,
+  encodeEd25519Key,
+  generateJWT,
+  jwtExp,
+} from "@walletconnect/did-jwt";
+import { ICore, IStore } from "@walletconnect/types";
+import { formatMessage, generateRandomBytes32 } from "@walletconnect/utils";
+import axios from "axios";
+import {
+  IIdentityKeys,
+  IdentityKeychain,
+  RegisterIdentityParams,
+  ResolveIdentityParams,
+  UnregisterIdentityParams,
+} from "./types";
+
+const DEFAULT_KEYSERVER_URL = "https://keys.walletconnect.com/";
+const IDENTITY_KEYS_STORAGE_PREFIX = "wc@2:identityKeys";
+
+export class IdentityKeys implements IIdentityKeys {
+  private keyserverUrl: string;
+  private identityKeys: IStore<IdentityKeychain["accountId"], IdentityKeychain>;
+
+  constructor(private core: ICore, keyServerUrl?: string) {
+    this.keyserverUrl = keyServerUrl ?? DEFAULT_KEYSERVER_URL;
+    this.identityKeys = new Store(
+      core,
+      this.core.logger,
+      "identityKeys",
+      IDENTITY_KEYS_STORAGE_PREFIX,
+      (keys: IdentityKeychain) => keys.accountId,
+    );
+  }
+
+  private generateIdentityKey = async () => {
+    const privateKey = ed25519.utils.randomPrivateKey();
+    const publicKey = await ed25519.getPublicKey(privateKey);
+
+    const pubKeyHex = ed25519.utils.bytesToHex(publicKey).toLowerCase();
+    const privKeyHex = ed25519.utils.bytesToHex(privateKey).toLowerCase();
+    this.core.crypto.keychain.set(pubKeyHex, privKeyHex);
+    return [pubKeyHex, privKeyHex];
+  };
+
+  public generateIdAuth = async (accountId: string, payload: JwtPayload) => {
+    const { identityKeyPub, identityKeyPriv } = this.identityKeys.get(`${accountId}_identityKeys`);
+
+    return generateJWT([identityKeyPub, identityKeyPriv], payload);
+  };
+
+  public async registerIdentity({ accountId, onSign }: RegisterIdentityParams): Promise<string> {
+    try {
+      const storedKeyPair = this.identityKeys.get(`${accountId}_identityKeys`);
+      return storedKeyPair.identityKeyPub;
+    } catch {
+      const [pubKeyHex, privKeyHex] = await this.generateIdentityKey();
+      const didKey = encodeEd25519Key(pubKeyHex);
+
+      const cacao: Cacao = {
+        h: {
+          t: "eip4361",
+        },
+        p: {
+          aud: this.keyserverUrl,
+          statement: "Test",
+          domain: this.keyserverUrl,
+          iss: composeDidPkh(accountId),
+          nonce: generateRandomBytes32(),
+          iat: new Date().toISOString(),
+          version: "1",
+          resources: [didKey],
+        },
+        s: {
+          t: "eip191",
+          s: "",
+        },
+      };
+
+      const cacaoMessage = formatMessage(cacao.p, composeDidPkh(accountId));
+
+      const signature = await onSign(cacaoMessage);
+
+      // Storing keys after signature creation to prevent having false statement
+      // Eg, onSign failing / never resolving but having identity keys stored.
+      this.identityKeys.set(`${accountId}_identityKeys`, {
```

Comment:
I think having the `_identityKeys` suffix on the key is redundant now that it's not a generic store but a concrete one owned by the `IdentityKeys` class. Using simply `accountId: string` as the index (as it is currently in the client implementations) should be sufficient now.

```suggestion
      this.identityKeys.set(accountId, {
```